internal/service/fleet.go

```diff
@@ -63,14 +63,16 @@ func (h *ServiceHandler) GetFleet(ctx context.Context, name string, params api.G
 	return result, StoreErrorToApiStatus(err, false, api.FleetKind, &name)
 }
 
-func (h *ServiceHandler) ReplaceFleet(ctx context.Context, name string, fleet api.Fleet) (*api.Fleet, api.Status) {
+func (h *ServiceHandler) ReplaceFleet(ctx context.Context, name string, fleet api.Fleet, isInternal bool) (*api.Fleet, api.Status) {
```

Comment:
Why are you passing isInternal explicitly rather than using IsInternalRequest(ctx)?

Reply:
Also make sure this bug is covered in our tests so it doesn't return.

Reply:
OK